### PR TITLE
[navigation]feat: add `sample data` menu back

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

In release 2.17 we won't make import sample data into a modal so we need to add the menu back.

## Screenshot

image

## Changelog

- feat: [navigation] Add `sample data` menu back

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/server/utils/auth_info.ts [hunk 1]
```diff
@@ -31,7 +32,9 @@ export const getPrincipalsFromRequest = (
     if (authState?.authInfo?.backend_roles) {
       payload[PrincipalType.Groups] = authState.authInfo.backend_roles;
     }
-    if (authState?.authInfo?.user_name) {
+    if (authState?.authInfo?.user_id) {
```
Comment (ruanyl) [inline src/core/server/utils/auth_info.ts:35]: This seems unrelated, is it by accident?

File: src/plugins/workspace/public/services/use_case_service.ts [hunk 2]
```diff
@@ -65,24 +65,32 @@ export class UseCaseService {
...
             },
             {
-              id: 'objects',
+              id: 'indexPatterns',
               category: DEFAULT_APP_CATEGORIES.manageWorkspace,
               order: 300,
             },
             {
-              id: WORKSPACE_DETAIL_APP_ID,
+              id: 'import_sample_data',
```
Comment (ruanyl) [inline src/plugins/workspace/public/services/use_case_service.ts:83]: Do we still have sample data card in overview pages?

File: src/core/server/utils/auth_info.ts (hunk 1 above)
Comment (SuZhou-Joe) [reply src/core/server/utils/auth_info.ts:35]: Nope, as the change is minor so I want to include the change into in this PR as well. No need to raise a new PR for such 1 line change.

File: src/plugins/workspace/public/services/use_case_service.ts (hunk 2 above)
Comment (SuZhou-Joe) [reply src/plugins/workspace/public/services/use_case_service.ts:83]: Yes, but we still need the menu in left nav or the bread crumb info will be broken.

**Review by ruanyl — APPROVED** [review]

File: src/plugins/workspace/public/services/use_case_service.ts (hunk 2 above)
Comment (AMoo-Miki) [reply src/plugins/workspace/public/services/use_case_service.ts:83]: I am having a hard time finding the Add Sample Data card in the workspaces other than Search. Is this something we added in the past day or two?

**Review by joshuali925 — APPROVED** [review]
